lib/worldwide/region.rb

```diff
@@ -435,13 +439,23 @@ def zip_autofill
 
     # is a postal code required for this region?
     def zip_required?
-      if zip_requirement.nil?
+      if @zip_requirement.nil?
         !zip_regex.nil?
       else
-        REQUIRED == zip_requirement
+        REQUIRED == @zip_requirement
       end
     end
 
+    # Returns whether (and how firmly) we require a value in the zip field
+    # Possible returns:
+    #  - "required" means a value must be supplied
+    #  - "recommended" means a value is optional, but we recommend providing one
+    #  - "optional" means a value is optional, and we say it is optional
+    # @return [String]
+    def zip_requirement
+      @zip_requirement || (zip_required? ? REQUIRED : OPTIONAL)
```

Comment:
If `zip_requirement` is not explicitly set on the region yaml, infer requiredness if the region defines a zip regular expression pattern. A region can define a pattern and set zip_requirement to "recommended" or "optional" to avoid this default.

Reply:
For my knowledge, what are the differences when setting `recommended` VS `optional`?